Refine e2e implementation

sdake opened the pull request:
Refine e2e implementation
    
 This is a refinement based upon the prior PRs in this area.
    
 Depends-On: https://github.com/istio/tools/pull/483

Commit:
WIP: Refine e2e implementation

This is a refinement based upon the prior PRs in this area.

Depends-On: https://github.com/istio/tools/pull/483

irisdingbj commented:
@sdake operator e2e already been fixed via https://github.com/istio/operator/pull/456. The reason why e2e fails in that PR is because it has not got the fix from https://github.com/istio/istio/pull/18352 .

sdake commented on Makefile.core.mk:
```diff
@@ -86,7 +86,7 @@ update-goldens:
 
 e2e:
-	@tests/e2e/e2e.sh
+	@HUB=$(HUB) TAG=$(TAG) bash -c tests/e2e/e2e.sh
```
Shell scripts will not take environment variables.  I tried HUB=$(HUB) TAG=$(TAG) ./tests/e2e/e2e.sh with no joy.  Quick Internet search shows this is the best solution to the problem.

linsun replied:
maybe put a comment here, it does look odd.

sdake commented:
@irisdingbj e2e did not work locally for me.  Additionally, there were several unaddressed comments - which should hopefully be resolved when this work merges.

But my apologies for not baackporting your work earlier and thinking e2e was broken in the checks.  Clearly an error on my part.  Would you submit a test-infra PR to make the check voting now?

Cheers
-steve

istio-testing commented:
@sdake: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e_operator | 629cc08af42532841878144d90a9baa9beacce30 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_operator/477/e2e_operator/141) | `/test e2e_operator`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@sdake: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e_operator | b86de6104d3af9467c228723106e1c89e014504a | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_operator/477/e2e_operator/142) | `/test e2e_operator`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@sdake: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e_operator | e2666f46d45a8d10ba8ad3b78acfcaf5663d1d1e | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_operator/477/e2e_operator/145) | `/test e2e_operator`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

Commit:
Use proper image with jq

sdake commented:
/test e2e_operator

incfly commented:
/cc @incfly

sdake commented:
@morvencao please take a look and approve if not already merged.  The gate is blocked.

Cheers
-steve
